pkg/validator/shoot_validator.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"errors"
+	"reflect"
+
+	"github.com/gardener/gardener-extension-provider-azure/pkg/apis/azure"
+
+	azurevalidation "github.com/gardener/gardener-extension-provider-azure/pkg/apis/azure/validation"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func (v *Shoot) validateShoot(shoot *core.Shoot, infraConfig *azure.InfrastructureConfig) error {
+	// Network validation
+	networkPath := field.NewPath("spec", "networking")
+
+	if errList := azurevalidation.ValidateNetworking(shoot.Spec.Networking, networkPath); len(errList) != 0 {
+		return errList.ToAggregate()
+	}
+
+	// Provider validation
+	fldPath := field.NewPath("spec", "provider")
+
+	if errList := azurevalidation.ValidateInfrastructureConfig(infraConfig, shoot.Spec.Networking.Nodes, shoot.Spec.Networking.Pods, shoot.Spec.Networking.Services, fldPath); len(errList) != 0 {
```

Comment:
shouldn't this be `infraConfigFldPath` instead? 

```suggestion
	if errList := azurevalidation.ValidateInfrastructureConfig(infraConfig, shoot.Spec.Networking.Nodes, shoot.Spec.Networking.Pods, shoot.Spec.Networking.Services, infraConfigFldPath); len(errList) != 0 {
```